### ISSUE TITLE
False "unexpected ')'" errors flagged

### ISSUE THREAD

**Issue by troyready** [AtomLinter/linter-erb#77]
I'm getting a `unexpected ')', expecting end-of-input` error reported from the linter that I can't reproduce manually.

Given a simple file test.sh.erb:

```
#!/bin/bash

if [ ! -f /home/<%= @username %>/Desktop/<%= @username.downcase %>file.img ]; then
  touch /home/<%= @username %>/Desktop/<%= @username.downcase %>file.img
fi
```

I can manually check it with `sed 's/<%=/<%/' test.sh.erb | erb -x - | ruby -c -` and it reports Syntax OK as expected. The linter in Atom reports the error on line 3 however.

The error disppears when I drop the `.downcase` methods, but I can't figure out why that's related. I suspect I'm reading the options passed to the commands incorrectly.

Let me know if there's some way I can help troubleshoot or resolve.

**Comment by Arcanemagus** [comment]
Do you have a trim mode set in the settings? That's the only thing that would be changing the output that I can see just looking at this. I'll test it in a bit.

**Comment by troyready** [comment]
No, I double checked and  my linter settings are all set to default.

**Comment by ricardograca** [comment]
Are you sure the file is being detected as a ERB file?

**Comment by troyready** [comment]
I think so, based on the error message:
![screen shot 2016-09-06 at 06 04 28](https://cloud.githubusercontent.com/assets/1806418/18274668/6a4cefe6-73f8-11e6-88cc-19504beef0d4.png)

The htmllinter also run against the file, but I think all of its warnings are prefixed differently.

**Comment by ricardograca** [comment]
Right, good call. It clearly says ERB. I'll be releasing a new version of this package soon, so hopefully that will take care of this issue since we had a few sort of similar issues. If that doesn't do it, then we'll try to investigate this further.

I'll update this issue when the new version is out.

**Comment by driehuis** [comment]
For what it's worth, I had the same error message on a index.html.erb file that validated just fine before I upgraded atom and all packages.

It was autodetected as HTML(Rails). At least, I do not recall manually selecting that, but I have troubles reproducing the autodetect behavior. As soon as I manually selected HTML(Ruby - ERB), the errors went away.

With HTML(Rails), it looks like this:
![screenshot from 2016-11-22 19 11 46](https://cloud.githubusercontent.com/assets/806186/20536127/27cc637c-b0e8-11e6-919a-4f686a87e6ec.png)

I'm happy with the workaround of manually selecting the type; I'm not (yet?) using Rails snippets.

**Comment by ricardograca** [comment]
This package (linter-erb) only lints files marked as `HTML(Ruby - ERB)`, so I don't think the error you're getting can be fixed on this side of the fence.

**Comment by driehuis** [comment]
That's weird. I cannot reproduce this behavior with linter-erb disabled, and I do not see any calls to erb -x in the language-ruby-on-rails package. Oh well, probably some weird interaction between packages that causes this. It's a heisenbug anyway. If it bites me again I'll try disabling other linters.

**Comment by ricardograca** [comment]
Do you have linter-htmlhint installed? That linter is known to lint files that it shouldn't, which results in all sorts of weird issues.

**Comment by Arcanemagus** [comment]
Can you update to v1.0.3 and check this? https://github.com/AtomLinter/linter-erb/pull/40 was released in that which might help here.

**Comment by driehuis** [comment]
It seems to be fixed after updating to v1.0.3. It used to trip over much code that was perfectly valid, as far as I can see, like this:

```
<%= search_form_for @search, url: search_hosts_path, html: {method: :post} do |f| %>
  Foo
<% end %>
```

**Comment by Arcanemagus** [comment]
Awesome, I'll mark this as closed then, please report back if this isn't fixed though!
